microservices/authentication/src/queries/services.py

```diff
@@ -0,0 +1,56 @@
+import base64
+import time
+
+import jwt
+from dependency_injector.wiring import (
+    Provide,
+)
+from minos.common import (
+    AggregateDiff,
+)
+from minos.cqrs import (
+    QueryService,
+)
+from minos.networks import (
+    Request,
+    Response,
+    RestRequest,
+    enroute,
+)
+
+from ..jwt_env import (
+    JWT_ALGORITHM,
+    SECRET,
+)
+from .repositories import (
+    UserQueryRepository,
+)
+
+
+class LoginQueryService(QueryService):
+    repository: UserQueryRepository = Provide["user_repository"]
+
+    @enroute.rest.query("/login", "GET")
+    async def get_token(self, request: RestRequest) -> Response:
+        auth_type, encoded_credentials = request.raw_request.headers["Authorization"].split()
+        if auth_type == "Basic":
+            user, password = base64.b64decode(encoded_credentials).decode().split(":")
+
+            if await self.valid_credentials(user, password):
+                jwt_token = await self.generate_token(user)
+                return Response(jwt_token)
+            else:
+                return Response("Invalid username or password")
+
+    async def generate_token(self, user):
+        payload = {"sub": 1, "name": user, "iat": time.time()}
```

Comment:
I would include another field to store the `User.uuid` to inject it over the internal `Request` calls.

Reply:
Probably we can replace the "name" field with that. I don't think we need the username in the token, right?

Reply:
Okay, I don't find any reason either hahaha